x/bank/commands/sendtx.go

```diff
@@ -0,0 +1,138 @@
+package commands
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/examples/basecoin/app"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	crypto "github.com/tendermint/go-crypto"
+	"github.com/tendermint/tmlibs/cli"
+)
+
+const (
+	flagTo       = "to"
+	flagAmount   = "amount"
+	flagFee      = "fee"
+	flagSequence = "seq"
+)
+
+// SendTxCommand will create a send tx and sign it with the given key
+func SendTxCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "send",
+		Short: "Create and sign a send tx",
+		RunE:  sendTx,
+	}
+	cmd.Flags().String(flagTo, "", "Address to send coins")
+	cmd.Flags().String(flagAmount, "", "Amount of coins to send")
+	cmd.Flags().String(flagFee, "", "Fee to pay along with transaction")
+	cmd.Flags().Int64(flagSequence, 0, "Sequence number to sign the tx")
+	return cmd
+}
+
+func sendTx(cmd *cobra.Command, args []string) error {
+	txBytes, err := buildTx()
+	if err != nil {
+		return err
+	}
+
+	uri := viper.GetString(client.FlagNode)
+	if uri == "" {
+		return errors.New("Must define which node to query with --node")
+	}
+	node := client.GetNode(uri)
+
+	result, err := node.BroadcastTxCommit(txBytes)
+	if err != nil {
+		return err
+	}
+
+	if result.CheckTx.Code != uint32(0) {
+		return errors.Errorf("CheckTx failed: (%d) %s",
+			result.CheckTx.Code,
+			result.CheckTx.Log)
+	}
+	if result.DeliverTx.Code != uint32(0) {
+		return errors.Errorf("CheckTx failed: (%d) %s",
+			result.DeliverTx.Code,
+			result.DeliverTx.Log)
+	}
+
+	fmt.Printf("Committed at block %d. Hash: %s\n", result.Height, result.Hash.String())
+	return nil
+}
+
+func buildTx() ([]byte, error) {
```

Comment:
@rigelrozanski  
this is tied to the tx struct, which can be different
for every app. How do we best keep most of the command logic in x/bank,
but allow overriding everything after buildMsg()?

I ask you, as you will be the first major user of this functionality
For now, we can just hard-code, but it is ugly.

Reply:
cool - I'll think about this some more - needs to constructed for applications to generically add their own structure